### ISSUE TITLE
iOS3

### ISSUE THREAD

**Issue by Burgch** [Stimpy5050/Twitter-Plugin-for-LockInfo#2]
NSRegularExpression is iOS4+ only, so this new plugin does not support iOS3, including iOS 3.2.

**Comment by ghost** [comment]
For that reason I was trying subString based approach but that was going complicated. May be to support IOS3 we will have to do it using subString only.

**Comment by Burgch** [comment]
I'm just curious, but what is the point of using HTML? Is a native code approach not favourable?

**Comment by ghost** [comment]
I think I accidentally closed this issue earlier.

**Comment by Burgch** [comment]
Oh yeah, I wondered why it disappeared after I submitted my comment. Anyways, webview?

**Comment by ghost** [comment]
You mean native detail view using UITextView? UITextView has limitations like it may only link detectable Data text (URL, Phone Number, Cal event etc.). For tweets we needed hashes and scree names to be linked as well so is HTML.

If we have native alternative for that please let me know. There is TTStyledText from Three20 but may be that will add more dependencies.

**Comment by Burgch** [comment]
I mean something like this: http://furbo.org/2008/10/07/fancy-uilabels/

**Comment by Burgch** [comment]
Though it might be easier to just add iOS3 support by using NSClassFromString to check for the presence of NSRegularExpression and if it doesn't exist use http://regexkit.sourceforge.net/RegexKitLite/ instead.

**Comment by ghost** [comment]
RegexKitLite is good alternative. Actually ours is not really that complex regex so we can easily do it using NSScanner or using simple String range/subString and will be efficient also I hope. I will give it a try.

**Comment by Burgch** [comment]
Though now you've got me thinking. You could probably roll a much better implementation than fancy-uilabels by doing your drawing manually. Parse the tweet using regex with NSClassFromString etc. to split it up, and then draw text with a for loop.

As you draw the text, add the rect of the links and their destination URL to an NSArray, implement pointInside to iterate through that array, and then implement touchesEnded to open the relevant link.

Though that is essentially rewriting UITextView, would probably be faster than creating a webview. I might look into it sometime.

**Comment by Burgch** [comment]
In response to your using subString approach, would it support multiple has tags and links in a tweet?

**Comment by ghost** [comment]
Yes, it does. I had it implemented actually but later I went ahead with NSRegularExpression. 

About our custom drawing to render links in text, yes that will be great. UIWebView sucks.

**Comment by Burgch** [comment]
Ahh excellent, so perhaps use your old implementation if NSClassFromString(@"NSRegularExpression"); returns nil?

Yeah, I'll look into it soon. The drawing and regex should be easy enough, the only trouble is touch handling. I think it's possible though.
